### DISCUSSION TITLE
Validation fails with "withInput does not exist."

### DISCUSSION THREAD

**Discussion by lasseeee** [livewire/livewire#5720] [7. Bugs (archived)] [answered]
I have installed Livewire on a fresh Laravel 10 install in PHP 8.2 as per the docs, and created this component:

```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class SomeComponent extends Component
{
    public $someDate;

    protected $rules = [
        'someDate' => 'nullable|date_format:d.m.Y',
    ];

    public function render()
    {
        $this->validate();

        return view('livewire.some-component');
    }
}
```

and this view `some-component.blade.php` view:

```
<div>
    <input wire:model="someDate" type="date">
</div>
```

and included in on default `welcome.php` like this:

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
<head>
    @livewireStyles
</head>
<body>
<livewire:some-component />
@livewireScripts
</body>
</html>
```

whenever I change the date I expect nothing to happen, as the validation should fail silently, but instead I get this error:
`Method Livewire\Redirector::withInput does not exist.`

<img width="713" alt="image" src="https://github.com/livewire/livewire/assets/1869424/b393a0a7-56b9-4d1e-9e23-4d9895a56d5a">

What am I missing?

**Comment by lasseeee** [comment]
I'd be so happy if anyone could try to replicate this on their machine by copy pasting the snippets above and letting me know if it works for them 🙏

**Comment by tomphilpotts** [comment, marked as answer]
The issue is because you have the $this->validate() within the render function. 

Try this instead:

```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class SomeComponent extends Component
{
    public $someDate;

    protected $rules = [
        'someDate' => 'nullable|date_format:d.m.Y',
    ];

    public function updatedSomeDate()
    {
        $this->validate();
    }

    public function render()
    {
        return view('livewire.some-component');
    }
}
```

**Reply by lasseeee** [reply]
Oh my, thank you! Makes complete sense🙏
